tuxemon/states/combat/combat.py

```diff
@@ -1199,6 +1199,28 @@ def trigger_player_run(self, player: NPC) -> None:
         self.players.remove(player)
 
     def evolve(self) -> None:
+        for monster in self.players[0].monsters:
+            for evolution in monster.evolutions:
+                if evolution.at_level <= monster.level:
+                    self.client.pop_state()
+                    tools.open_dialog(
+                        local_session,
+                        [
+                            T.format(
+                                "evolution_confirmation",
+                                {"name": monster.name},
+                            )
+                        ],
+                    )
+                    tools.open_choice_dialog(
+                        local_session,
+                        menu=(
+                            ("yes", T.translate("yes"), self.positive_answer),
```

Comment:
You need to define `positive_answer` within this method.
```
def positive_answer():
    self.players[0].evolve_monster(
        monster, evolution.monster_slug
    )
```
You can also use a library function: functions.partial, but this might be too complicated. Anyway, both of these ways are used frequently in the project.

Reply:
Ok, I'll look into